plugin/pkg/admission/serviceaccount/admission.go

```diff
@@ -180,6 +182,7 @@ func (s *serviceAccount) Admit(a admission.Attributes) (err error) {
 		}
 	}
 
+	s.mountVolumeForTokenRequest(pod, a)
```

Comment:
needs to be gated based on the feature-gate and the apiserver surfacing the token request API endpoint

Reply:
AutomountServiceAccountToken should apply whether we are mounting a secret or a projection volume. I would not expect to inject both a secret and projection volume, only one or the other